lib/tasks/marks.rake

```diff
@@ -21,9 +13,14 @@ namespace :db do
       grouping.save
 
       #Automate marks for assignment using appropriate criteria
-      grouping.assignment.get_criteria.each do |criterion|
-        mark = create_mark(result.id, criterion)
-        result.marks.push(mark)
+      grouping.assignment.get_criteria(:all, :all, includes: :marks).each do |criterion|
+        mark = Mark.find_by(result_id:     result.id,
+                            markable_id:   criterion.id,
+                            markable_type: criterion.class.to_s)
+        mark.update({ result_id:     result.id,
+                      mark:          rand(0..criterion.max_mark.to_i),
```

Comment:
I see what you mean. criterion.max_mark would work for checkbox and flexible because criterion.max_mark would actually be the maximum mark a marker can give. But in the rubric case criterion.max_mark, the maximum mark a marker can give is 4. I will change this.